### DISCUSSION TITLE
disallow player login

### DISCUSSION THREAD

**Discussion by GoldenCookieGIT** [Minestom/Minestom#1314] [Help] [answered]
How would I go about disallowing players to log in? The player login event isnt cancellable and even if i kick the player it still tells me i should've set a spawning instance

**Comment by ZakShearman** [comment, marked as answer]
Perhaps you're looking for AsyncPlayerPreLoginEvent or do you need to cancel the login at a later point?

**Reply by GoldenCookieGIT** [reply]
that did it, was looking for a PlayerPreLoginEvent but intellij could find one :grief:
